scripts/console.ts

```diff
@@ -33,25 +36,29 @@ import {
     deployTokenFactory,
 } from "./deploy";
 import { saltToHex } from "./utils";
-import { GovernorFactory, TokenFactory } from "../types";
-import { GovernorImpl, TokenImpl } from "../types/contracts/clones";
-
-// eslint-disable-next-line @typescript-eslint/no-var-requires, node/no-unpublished-require
-let deployments: Deployments = require(`../deployments.json`);
+import { VERBOSE } from "../hardhat.config";
 
 // --- Provides a CLI to deploy the possible contracts ---
 
 // TODO: Hardware Wallet support:
 // https://docs.ethers.io/v5/api/other/hardware/
 
-const JSON_NUM_SPACES = 4;
+// undefined gas amounts will be estimated at transaction creation time
+let gasPrice: BigNumber | undefined;
+let gasLimit: BigNumber | undefined;
 
 async function main(signer?: SignerWithAddress): Promise<void> {
     if (signer == undefined) {
         signer = await askForSigner();
     }
-
-    // TODO: separate out switches into their own function
+    if (VERBOSE) {
```

Comment:
Semantically, it seems weird to me that we're using `VERBOSE`, which usually signifies additional logging, to allow users to provide additional arguments to the transaction call. Someone may want to provide a gas price and gas limit but not want verbose logging. Could we add an additional environmental variable (e.g., `CONFIG_GAS`)? We would also need to caution users in setting the gas limit too low, as this may result in the transaction being unfulfilled for an indefinite amount of time.

Reply:
That makes a lot of sense, though I have some concerns about the env vars getting too bloated. I pushed a change that uses a more generic `GAS_MODE` that replaces the previous `REPORT_GAS` + adds this configuration. 

I figure this is a good middle ground that keeps the env vars slim, but should be clear for users looking to deploy with custom gas settings and additional reporting. Also added comment about it in hardhat config.

Let me know what you think of this approach or if it makes more sense to you to just add a new variable.

Reply:
Makes sense to me, as long as there's a way for users wanting to report gas but not provide custom gas settings (which there is with the sentinel `0` gas price/limit).